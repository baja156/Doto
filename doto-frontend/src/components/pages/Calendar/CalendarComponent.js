import React from "react";
import "./Calendar.css";
import { ViewState } from "@devexpress/dx-react-scheduler";
import PropTypes from "prop-types";
import {
    Scheduler,
    WeekView,
    MonthView,
    DayView,
    ViewSwitcher,
    Toolbar,
    DateNavigator,
    TodayButton,
    Appointments,
    AppointmentTooltip,
} from "@devexpress/dx-react-scheduler-material-ui";

const CalendarComponent = props => {
    return (
        <Scheduler data={props.tasks} currentView={MonthView} editable={true}>
            <ViewState />
            <DayView />
            <WeekView />
            <MonthView />
            <Toolbar />
            <ViewSwitcher />
            <DateNavigator />
            <TodayButton />
            <Appointments />
            <AppointmentTooltip />
        </Scheduler>
    );
};

CalendarComponent.propTypes = {
    tasks: PropTypes.array.isRequired,
};

export default CalendarComponent;
