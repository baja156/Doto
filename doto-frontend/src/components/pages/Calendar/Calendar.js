import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Modal from "@material-ui/core/Modal";
import Backdrop from "@material-ui/core/Backdrop";
import Fade from "@material-ui/core/Fade";
import ModalConent from "../../ModalContent";
import CalendarComponent from "./CalendarComponent";
import CalendarListView from "./CalendarListView";
import Fab from "@material-ui/core/Fab";
import FormatListBulletedIcon from "@material-ui/icons/FormatListBulleted";
import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
import AddIcon from "@material-ui/icons/Add";
import Tooltip from "@material-ui/core/Tooltip";
import Header from "../Header";
import "./Calendar.css";
import "../Pages.css";

const useStyles = makeStyles(theme => ({
    modal: {
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
    },
    paper: {
        backgroundColor: theme.palette.background.paper,
        border: "2px solid #000",
        boxShadow: theme.shadows[5],
        padding: theme.spacing(2, 4, 3),
    },
}));

const Calendar = () => {
    const classes = useStyles();
    const [listView, setListView] = useState();
    const [appointments, setAppointments] = useState([]);
    const [open, setOpen] = useState(false);

    const handleOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    useEffect(() => {
        const tempAppointments = [
            {
                title: "Lecture-1",
                descrition: "Lecture at Auckland Uni",
                startDate: new Date(2020, 2, 12, 9, 0), // 0 is January
                endDate: new Date(2020, 2, 12, 11, 0),
                id: 0,
                priority: 3,
                location: "Room 1",
                color: "blue",
            },
            {
                title: "Lecture-2",
                descrition: "Lecture at Auckland Uni",
                startDate: new Date(2020, 2, 10, 10, 0),
                endDate: new Date(2020, 2, 10, 12, 0),
                id: 1,
                priority: 2,
                location: "Room 2",
                color: "green",
            },
            {
                title: "James' Birthday",
                descrition: "James' day of birth",
                startDate: new Date(2020, 2, 13, 0, 0),
                endDate: new Date(2020, 2, 14, 0, 0),
                id: 2,
                priority: 1,
                location: "Room 3",
                color: "pink",
            },
        ];

        setAppointments(tempAppointments);
    }, []);

    return (
        <div className="PageLayout">
            {/* Wrong purple will change in next PR */}
            <div className="left-side-bar bg-purple-800" />
            {/* className={classnames("left-side-bar", theme ? "left-side-bg-blue" : "left-side-bg-green")} /> */}
            <div className="calendar-buttons">
                <div className="mb-3">
                    <Tooltip title="Add Task">
                        <Fab onClick={handleOpen} size="small">
                            <AddIcon />
                        </Fab>
                    </Tooltip>
                </div>
                <div className="mb-3">
                    <Tooltip title="List View">
                        <Fab onClick={() => setListView(!listView)} size="small">
                            {!listView && <FormatListBulletedIcon />}
                            {listView && <CalendarTodayIcon />}
                        </Fab>
                    </Tooltip>
                </div>
            </div>
            <span className="content-container">
                <Header title="Calendar" />
                <div className="flex">
                    <div className="Calendar">
                        <CalendarComponent appointments={appointments} />
                    </div>
                    {listView && <CalendarListView appointments={appointments} />}
                </div>
                <Modal
                    aria-labelledby="transition-modal-title"
                    aria-describedby="transition-modal-description"
                    className={classes.modal}
                    open={open}
                    onClose={handleClose}
                    closeAfterTransition
                    BackdropComponent={Backdrop}
                    BackdropProps={{
                        timeout: 500,
                    }}
                >
                    <Fade in={open}>
                        <div className={classes.paper}>
                            <ModalConent />
                        </div>
                    </Fade>
                </Modal>
            </span>
        </div>
    );
};

export default Calendar;
