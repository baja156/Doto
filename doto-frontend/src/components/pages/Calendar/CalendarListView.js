import React from "react";
import Radio from "@material-ui/core/Radio";
import "./Calendar.css";
import PropTypes from "prop-types";

const CalendarListView = props => {
    const today = new Date();

    return (
        <div className="list-view">
            <div className="ml-3 mb-5 text-4xl font-bold">Tasks for Today</div>

            {props.tasks.map(task => {
                const dueDate = new Date(task.endDate);
                const isTaskScheduledToday =
                    dueDate.getYear() === today.getYear() && dueDate.getMonth() === today.getMonth()
                        ? dueDate.getDate() === today.getDate()
                        : false;

                return (
                    isTaskScheduledToday && (
                        <div key className="list-view-components">
                            <Radio
                                disableRipple
                                checkedIcon={
                                    <span
                                        className={"radio-checked bg-" + (task.color ? task.color : "blue") + "-600"}
                                    />
                                }
                                icon={<span className="radio-unchecked" />}
                            />
                            <div className="text-base">{task.title}</div>
                        </div>
                    )
                );
            })}
        </div>
    );
};

CalendarListView.propTypes = {
    tasks: PropTypes.array.isRequired,
};

export default CalendarListView;
