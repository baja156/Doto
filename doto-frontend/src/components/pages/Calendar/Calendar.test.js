import React from "react";
import ReactDOM from "react-dom";
import { mount, shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import FormatListBulletedIcon from "@material-ui/icons/FormatListBulleted";
import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
import Calendar from "./Calendar";
import CalendarListView from "./CalendarListView";
import { ThemeContext } from "../../../context/ThemeContext";

describe("<Calendar /> component being rendered", () => {
    let theme;
    const setTheme = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(theme => [theme, setTheme]);

    const Wrapper = () => {
        return (
            <Router>
                <ThemeContext.Provider value={[theme, setTheme]}>
                    <Calendar />
                </ThemeContext.Provider>
            </Router>
        );
    };

    beforeEach(() => {
        useStateSpy.mockImplementation(theme => [theme, setTheme]);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("Calendar component rendered without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Wrapper />, div);
    });

    it("Make sure render matches snapshot", () => {
        const tree = shallow(<Wrapper />);
        expect(tree.debug()).toMatchSnapshot();
        tree.unmount();
    });

    it("Click on List View button should open CalendarListView while changing its icon", () => {
        const subject = mount(<Wrapper />);

        const button = () => {
            // search for List View button
            return subject.find('button[title="List View"]');
        };

        expect(subject.find(CalendarListView)).toHaveLength(0);
        expect(button().find(FormatListBulletedIcon)).toHaveLength(1);
        expect(button().find(CalendarTodayIcon)).toHaveLength(0);

        button().simulate("click");

        expect(subject.find(CalendarListView)).toHaveLength(1);
        expect(button().find(FormatListBulletedIcon)).toHaveLength(0);
        expect(button().find(CalendarTodayIcon)).toHaveLength(1);

        subject.unmount();
    });
});
