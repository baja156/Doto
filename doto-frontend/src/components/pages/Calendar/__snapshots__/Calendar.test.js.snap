// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> component being rendered Make sure render matches snapshot 1`] = `
"<BrowserRouter>
  <ContextProvider value={{...}}>
    <Calendar />
  </ContextProvider>
</BrowserRouter>"
`;
